### PR TITLE
Adding support for mice models (from multiple imputation)

### LINKED ISSUE #8: Add a `model_get_model_frame()` method for mice objects
The mice package does not include a `model.frame()` method for the resulting regression models from multiply imputed data sets.

Would you be ok adding one here?  I need to look up the exact code, but it'll be something like this (i can add if you're ok with it)

```r
#' @export
#' @rdname model_get_model_frame
model_get_model_frame.mipo <- function(model) {
  # add check that the mice package is installed
  
  # grab input mice data

  # extract a single dataset for our use of finding labels
   mice::complete(...)
}
```

### PR DESCRIPTION
**Pull request by ddsjoberg**
This PR adds support for mice models.  The mice package contains its own tidier function `mice::tidy()`. 

If there are 10 imputed sets of data, there are 10 separate models that were estimated (before being combined into the final single result).  What is being done here is the 1st model being estimated on the 1st imputed set of data is extracted and added as an attribute to the tidy data frame.

The numbers in the extracted model object are not the final results, but the model contains all the information about the structure, variables, etc.

I haven't added any unit tests yet, and this should be done before merging. I first wanted to ask how you wanted this organized. In gtsummary, we have a list of model types we refer to as vetted models. These are models we have tested, and are confident work with the current structure to produce results.  We came up with a list of checks that each vetted model must pass.

closes #8

### CODE REVIEW

File: R/tidy_add_reference_rows.R [hunk 1]
```diff
@@ -186,11 +186,11 @@ tidy_add_reference_rows <- function(x, model = tidy_get_model(x)) {
   }
 
   if (!has_var_label)
-    x <- x %>% dplyr::select(-.data$var_label)
+    x[["var_label"]] <- NULL
```
Comment (ddsjoberg) [inline R/tidy_add_reference_rows.R:189]: FYI, I made this update because the model attribute was being stripped from the tidy tibble.

Comment (larmarange) [reply R/tidy_add_reference_rows.R:189]: Thanks a lot for your proposal and more in general for testing the different features. Do you mind to be added as a contributor?

Regarding mice objects, I'm not sure that tidy_and_attach is the good entry point as the model could be unattached and passed explicitly to all functions.

Would it be better to attach the full mice object and then to add specific methods for the model_* functions?

**Comment by larmarange** [comment]
By the way, if the model attribute is lost in some cases, should we add a call to tidy_attach_model at the end of each tidy_* function, by security?

File: R/tidy_add_reference_rows.R (hunk 1 above)
Comment (ddsjoberg) [reply R/tidy_add_reference_rows.R:189]: That makes sense.  There is already a method fn for model.frame.  I'll need to add one for `stats::terms()` and the fn that gets the contrasts as well.

Happy to be a contributor here.  It's an awesome project!

**Comment by ddsjoberg** [comment]
> By the way, if the model attribute is lost in some cases, should we add a call to tidy_attach_model at the end of each tidy_* function, by security?

I think that is a good idea. A simple check if the model attr is still there, if not, add it back?

**Comment by ddsjoberg** [comment]
@larmarange 

Oof, so I updated to extract the model whenever it's needed.
1.  The `get_model_mipo()` is _not_ robust, and fails whenever the user created the mice (mipo) object in a non-standard way (e.g. with apply, map, or even piping).
1. The way this is written, the model will be built many many times. If they model is even remotely large, this will waste a lot of computation time.
1. Since the mice regression model objects are so different from a typical model, I am re-thinking and likely the package should _not_ support them.
 
I can still use the package for these model objects by supplying my own model object after performing initial tidying.

**Comment by larmarange** [comment]
OK well noted.

Maybe you could discuss with the authors of mice models about the relevance of adding model.frame and model.matrix methods in their package.

**Pull request closed without merging**